server/game/modules/deal.js

```diff
@@ -0,0 +1,58 @@
+/**  Function Should:
+ * run whenever a deal action comes into the receiver
+ * create a deck based on the deck parameters
+ * deal a card to each player
+ * -->get a random card from the deck
+ * -->shuffle the deck (for fun)
+ * -->assign that card to the player by updating the player table
+ * -->repeat for all players
+ * emit socket events to update the client(s)
+*/
+
+//the structure for what a deal action will look like from the client
+const sampleDealAction = {
+    type: 'deal',
+    data: {
+       roundNumber: '1', //round number as a string from client redux state
+    },
+    facilitatorId: 0,
+}
+
+
+//Deck class contains all methods required for dealing 
+const Deck = require('./Deck');
+const pool = require('../../modules/pool');
+
+const deal =  async (action, gameId, socket) => {
+    try {
+        //get cards
+       let cardResponse  = await pool.query(`SELECT * FROM "card";`)
+       //get players
+       let playerResponse = await pool.query(`SELECT "id" FROM "player" WHERE "game_id"=$1;`, [gameId])
+       //set cards as card data
+       let cards = cardResponse.rows;
+       //set players as players data
+       let players = playerResponse.rows;
+       //create a deck object with Deck class
+       let deck =  new Deck(cards);
+       //set selected deck by round number
```

Comment:
deck.shuffle() should go here, to shuffle before anything is drawn or even sub selected so everything is random from the start